src/bread-n-butter.ts

```diff
@@ -307,6 +307,57 @@ export function match(regexp: RegExp): Parser<string> {
   });
 }
 
+/** Parse all items, returning their values in the same order. */
```

Comment:
There is a better way to write this. Instead of all the overrides, use a generic rest type:

```typescript
type Parserify<A extends any[]> = {
  [P in keyof A]: Parser<A[P]>;
};

export function all<A extends any[]>(...parsers: Parserify<A>): Parser<A> {
  return parsers.reduce((acc, p) => {
    return acc.chain((array) => {
      return p.map((value) => {
        return [...array, value];
      });
    });
  }, ok([]));
}
```

This will change the signature from `all([parserA, parserB, parserC])` to `all(parserA, parserB, parserC)`, which seems simpler to me, and it will keep the signature of each parser, whatever the number of them.

Successfully tested using your own test suite on my computer.

Reply:
I tested this without using spread `...` arguments and it _does_ work, but you have to use `readonly any[]` **AND** use `as const` after your array in `bnb.all`, so I definitely prefer your approach!